eZ/Publish/API/Repository/Values/Content/Field.php

```diff
@@ -49,4 +50,11 @@ class Field extends ValueObject
      * @var string
      */
     protected $languageCode;
+
+    /**
+     * Field type identifier.
+     *
+     * @var string
+     */
+    protected $type;
```

Comment:
More of a nitpick perhaps, as naming is hard: But we have a lot of `type`'s in the system, maybe align with FieldDefinition rather to make sure it's clear what this is and avoid introducing different name for same thing?  Alternatively rename field _(deprecate old and add new)_ in FieldDefinition so we have consistent naming. Like for instance `->typeIdentifier`, which can be confused by `$type->identifier` (: 

ref:
```
    /**
     * String identifier of the field type.
     *
     * @var string
     */
    protected $fieldTypeIdentifier;
```

cc @bdunogier

Reply:
Hmm, ok so we need @bdunogier's decision, because I've taken `type` from [EZP-28276](https://jira.ez.no/browse/EZP-28276) issue description :)

Reply:
Ok, from my side `fieldTypeIdentifier` seems to be the better choice, it's used in REST _(incl added here)_ and PHP API.

Reply:
Even though it will mean repetitions of the word "field" in the code (`if ($field->fieldTypeIdentifier === $someFieldTypeIdentifier)`), I see the point about consistency. 

I'd still be tempted to suggest `$field->typeIdentifier`, and to be consistant from now on in this regard. It's the same than `$content->id`. But if you insist, you can have my +1 for the safe, full-length version.

Yes, naming is hard... :)

Reply:
`$field->typeIdentifier` is okish, as long as we then cleanup FieldDefinition _(deprecate old, and add new)_ as well.

Given we are both almost ok with both cases, `fieldTypeIdentifier ` or `typeIdentifier`, @alongosz you have options to pick from, if unsure check with others on the teams what they prefer ;)

Reply:
![watchingsadly](https://user-images.githubusercontent.com/5822569/33321272-67daa63c-d445-11e7-98d3-ab64aac84619.gif)

(But vote for `typeIdentifier` :P)

Reply:
`$field->typeIdentifier` sounds like a good compromise. I can use it both here and in REST response for consistency.

However please **do not make me right now** deprecate the FieldDefinition (`fieldDefIdentifier`). It is out of scope and it is a sprint blocker for 2 front-end features.
It can be a follow-up.